sdks/python/apache_beam/transforms/window.py

```diff
@@ -190,12 +192,17 @@ def __init__(self, end):
   def max_timestamp(self):
     return self.end.predecessor()
 
-  def __cmp__(self, other):
-    # Order first by endpoint, then arbitrarily.
-    return cmp(self.end, other.end) or cmp(hash(self), hash(other))
-
   def __eq__(self, other):
-    raise NotImplementedError
+    return type(self) is type(other) and self.end == other.end
+
+  def __ne__(self, other):
+    return not self.__eq__(other)
+
+  def __lt__(self, other):
```

Comment:
I think a type mismatch should return `NotImplemented` here, or raise an error.

Reply:
Sounds reasonable, I'll try that.